sockeye/train.py

```diff
@@ -724,24 +689,42 @@ def define_optimizer(args, lr_scheduler_instance) -> Tuple[str, Dict, str, str,
     # Manually specified params
     if args.optimizer_params:
         optimizer_params.update(args.optimizer_params)
-    logger.info("Optimizer: %s", optimizer)
-    logger.info("Optimizer Parameters: %s", optimizer_params)
-    logger.info("kvstore: %s", args.kvstore)
-    logger.info("Gradient Compression: %s", gradient_compression_params(args))
 
-    return optimizer, optimizer_params, args.kvstore, gradient_clipping_type, gradient_clipping_threshold
+    config = OptimizerConfig(name=args.optimizer,
+                             params=optimizer_params,
+                             kvstore=args.kvstore,
+                             initializer=initializer.get_initializer(
```

Comment:
I would separate the creating of the initializer and the lr_scheduler into previous calls, for clarity.

Reply:
ok